infra/Pos/NtpCheck.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Pos.NtpCheck
+    ( withNtpCheck
+    , NtpStatus(..)
+    ) where
+
+import           Universum                   hiding (bracket)
+
+import           Control.Monad.Trans.Control (MonadBaseControl)
+import qualified Data.List.NonEmpty          as NE
+import           Data.Time.Units             (Microsecond)
+import           Mockable                    (Bracket, CurrentTime, Fork, Mockable,
+                                              bracket, currentTime)
+import           NTP.Client                  (NtpClientSettings (..), pressNtpStopButton,
+                                              startNtpClient)
+import           Pos.Core.Timestamp          (Timestamp (..), diffTimestamp)
+import           Pos.Infra.Configuration     (HasInfraConfiguration, infraConfiguration)
+import qualified Pos.Infra.Configuration     as Infra
+import           Pos.Util.Util               (median)
+import           Serokell.Util               (sec)
+import           System.Wlog                 (WithLogger)
+
+type NtpCheckMonad m =
+    ( MonadIO m
+    , MonadMask m
+    , MonadBaseControl IO m
+    , Mockable Fork m
+    , Mockable Bracket m
+    , Mockable CurrentTime m
+    , WithLogger m
+    , HasInfraConfiguration
+    )
+
+withNtpCheck :: forall m a. NtpCheckMonad m => (NtpStatus -> m ()) -> m a -> m a
+withNtpCheck onStatus action = bracket (startNtpClient ntpSettings) pressNtpStopButton (\_ -> action)
+  where
+    ntpSettings :: NtpClientSettings m
+    ntpSettings = NtpClientSettings
+        { ntpServers         = Infra.ntpServers
+        , ntpHandler         = ntpCheckHandler onStatus
+        , ntpLogName         = "ntp-check"
+        , ntpResponseTimeout = sec 1
```

Comment:
That's the value that `ntpSingleShot` uses. If we increase it here, should we increase it there as well?

Reply:
`mkNtpSlottingVar` (which has `ntpResponseTimeout` set to 1 sec) is used only once on start because we don't want start-up to be slow. Then we use timeout from configuration which is 7 sec.
Here we don't wait for `ntpResponseTimeout`, so start-up shouldn't be slow if you increase it to 5 seconds.

Reply:
Forget it, I don't know why I thought `ntpSingleShot` was using 1s. I'll change `ntpResponseTimeout` to be the same as `timeDifferenceWarnThreshold`.